src/core/Segment/Segment.h

```diff
@@ -1,5 +1,71 @@
 #pragma once
+#include <algorithm>
+#include <cmath>
+#include <memory>
+#include <stdexcept>
+#include <utility>
+#include <vector>
 
-class Segment {
+#define PRECISION 0.001
+class Point {
+public:
+    double x, y;
+    explicit Point(double a = 0, double b = 0) {
+        x = a, y = b;
+    }
+    double norm() const { return std::sqrt(std::pow(x, 2) + std::pow(y, 2)); }
+    Point operator+(Point b) { return Point(x + b.x, y + b.y); }
+    Point operator-(Point b) { return Point(x - b.x, x - b.y); }
+    bool operator==(Point b) const {
+        return (std::abs(x - b.x) < PRECISION && std::abs(y - b.y) < PRECISION);
+    }
 };
```

Comment:
Perhaps, you might be worth creating separate description/implementation files of the `Point` class (`Point.h`/`Point.cpp`).